test/executor.test.js

```diff
@@ -317,44 +331,84 @@ describe('executor', function() {
         done();
       });
     });
+  });
 
-    describe('for mixins', function() {
-      var options;
-      beforeEach(function() {
-        appdir.writeFileSync('custom-mixins/example.js',
-          'module.exports = ' +
-          'function(Model, options) {}');
-
-        appdir.writeFileSync('custom-mixins/time-stamps.js',
-          'module.exports = ' +
-          'function(Model, options) {}');
+  describe('with boot script returning a rejected promise', function() {
+    before(function() {
+      // Tell simple-app/boot/reject.js to return a rejected promise
+      process.rejectPromise = true;
```

Comment:
It would be probably simpler to create a different application (a set of instructions) for each of these error cases. Having said that, your approach with process flags is ok too.